index.js

```diff
@@ -159,18 +200,18 @@ Filter.prototype.processFile =
   var outputEncoding = this.outputEncoding;
   if (inputEncoding === void 0) inputEncoding = 'utf8';
   if (outputEncoding === void 0) outputEncoding = 'utf8';
-  var contents = fs.readFileSync(
-      srcDir + '/' + relativePath, { encoding: inputEncoding });
 
-  return Promise.resolve(this.processString(contents, relativePath)).
-      then(function asyncOutputFilteredFile(outputString) {
+
+  var contents = this.in.readFileSync(relativePath, {encoding : inputEncoding});
```

Comment:
we want to `this.in[0].readFileSync` not `this.in.readFileSync`.  If you look at the previous version you'll see the `srcDir` is `this.inputPaths[0]`.  `this.in` is a merge tree which also responds to `0`, `1`, &c. for each of the input paths.  Much of the time plugins only care about their first input, in which case we want `this.in[0]`.

It won't work with the current shim you have b/c it makes an `FSTree` instead of an `FSMergeTree` but i'll push a commit in just a bit that updates this branch to use the `in` and `out` from broccoli-plugin